src/main/java/com/whiteclarkegroup/liquibaselinter/report/ReporterConfig.java

```diff
@@ -0,0 +1,83 @@
+package com.whiteclarkegroup.liquibaselinter.report;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import static com.whiteclarkegroup.liquibaselinter.report.ReportItem.ReportItemType.ERROR;
+import static com.whiteclarkegroup.liquibaselinter.report.ReportItem.ReportItemType.IGNORED;
+
+@JsonDeserialize(builder = ReporterConfig.Builder.class)
+public class ReporterConfig {
+    private final boolean enabled;
+    private final String path;
+    private final Set<ReportItem.ReportItemType> filter;
```

Comment:
I'm not so sure that is true. This change introduces passing rules to the report. I might want to display those passing results in my markdown report. With what you suggest, I'd have to make a custom markdown reporter just to print the passing rules.

And for that matter, we might as well eliminate the whole concept of passing rules for this PR as there will be no reporter that can display them. 

My preference is to keep this configuration item. The number of lines supporting the feature are rather insignificant.

Reply:
Also, by eliminating this configuration option, then there is no reason to allow multiple reporters of the same type as the only real configuration option is where to write the report. There is no point in writing the same report multiple times to different locations. I would just perform file copies in my CI process.

Reply:
> And for that matter, we might as well eliminate the whole concept of passing rules for this PR as there will be no reporter that can display them.

Exactly, I can't see a case for tracking passing results really. For a test runner, sure - you want the feedback about all the tests. For a linter though? Linters are more like "no news is good news", and I think this is reflected in what's out there in other tools. If we report passes what will that look like? The number of changes/changesets/changelogs multiplied by the number of active rules...I don't think that data is valuable.

> Also, by eliminating this configuration option, then there is no reason to allow multiple reporters of the same type as the only real configuration option is where to write the report.

Right, good point, so the reporting config could be simplified to something like:

```json
{
  "console": true,
  "markdown": "target/lqlint-report.md"
}
```

I think that would be good.

Reply:
The code you just provided is exactly how it works with my code right now. 

| I can't see a case for tracking passing results really

There was only one reason why I brought in the passing results, and that is because I was pointed at the [junit-reporter](https://github.com/whiteclarkegroup/liquibase-linter/compare/junit-reporter) branch. If you don't want the junit-reporter to report passing lint rules, then I can remove that.

As for removing all notion of filtering, I'll conceed that there is not strong value to configure that for a JUnit reporter and I agree that pretty much every other reporter will likely have the same *default* filter that they want applied. When it comes time to actually make the JUnit reporter, we can make sure the filter cannot be used, if that is what you want.

However, if I eliminate this `filter` config option, then I will be unable to do what got me started down this whole path in making this PR. I want 2 markdown reports. Quality Control Engineers want to see `IGNORED` violations to know if dev's are inappropriately ignoring violations, but the DBAs are not so interested in the `IGNORED` violations and don't want that noise cluttering up their report. Removing the filter forces me to make a new reporter, stick it in a JAR and make sure I include it in my `pom.xml`

| We could always relook at adding some kind of generic configurability for reporters later but I think we should simplify for now.

Why not do it now since I have it working? Let's say I want to write a new reporter that pushes a reports to, for example, a Google Cloud Storage Bucket. I might need to provide some credentials or account information to make this work. Or maybe I want to publish violations to some database table somewhere. I would need some connection information specified for my reporter. 

By eliminating the ability to have custom config for reporters (and rules), we really limit what people can do. Users will be much more inclined to dismiss this liquibase linter due to the inflexibility for configuration.

Reply:
> There was only one reason why I brought in the passing results, and that is because I was pointed at the junit-reporter branch. If you don't want the junit-reporter to report passing lint rules, then I can remove that.

Thanks, yeah I think we can and should remove passes from being reported then.

Reply:
> By eliminating the ability to have custom config for reporters (and rules), we really limit what people can do. Users will be much more inclined to dismiss this liquibase linter due to the inflexibility for configuration.

Maximum users isn't the goal here - can't be all things to all people :)

That said I'd want things to be _possible_ - I would just tend to lean more to extension than configuration. I think some aspects of what you want to do (e.g. subtly different versions of the same report for different audiences) are on the niche side but it's nonetheless instructive in how we make it possible e.g. the need to potentially use the same reporter multiple times (which my example config structure above wouldn't allow for).

> Removing the filter forces me to make a new reporter, stick it in a JAR and make sure I include it in my pom.xml

I suppose I wouldn't see this as a big deal. We should look to make this as easy as possible in terms of documentation, base classes to extend (per another thread here), maybe an example project (lotta overlap with custom rules too).

Going to dig into this a little more today and circle back. I think there is a good case for some generic configurability of reporters.